### ISSUE TITLE
Skaffold is not injecting the autogenerated dateTime tag into Helm values

### ISSUE THREAD

**Issue by none-da** [GoogleContainerTools/skaffold#2856]
### Expected behavior
Skaffold should inject the generated dateTime tag into Helm's values properly.

### Actual behavior
Skaffold is not injecting the generated dateTime tag into Helm's values, but instead taking the digest value always.

### Information

- Skaffold version: v0.37.1
- Operating system: MacOS Mojave 10.14.6
- Contents of skaffold.yaml:

```yaml
apiVersion: skaffold/v1beta13
kind: Config
build:
  artifacts:
  - image: local/hello-universe
    context: hello-universe-flask-app
  tagPolicy:
    dateTime:
      timezone: "Local"
deploy:
  statusCheckDeadlineSeconds: 10
  helm:
    releases:
    - name: v1
      chartPath: "hello-universe-chart"
      wait: true
      recreatePods: true
      values:
        image: local/hello-universe
```

### Steps to reproduce the behavior

1. a clonable repository with the sample skaffold project : https://github.com/none-da/test_skaffold_helm
2. `skaffold <command>` : skaffold run  --tail --skip-tests -v debug
3. Prints:

```
Generating tags...
 - local/hello-universe -> local/hello-universe:2019-09-11_17-23-19.057_IST
```

But the command that is fed to helm is:

```
DEBU[0000] Running command: [helm --kube-context docker-desktop install --name v1 hello-universe-chart --wait --set image=local/hello-universe:59268a9ef3ecf2e38590b917985bc0a75711a304615d9522b5cc3cab5ee4b8d4]
```

**Comment by none-da** [comment]
Issue is still there in the recently released v0.39.0 version. 👇 
```Generating tags...
 - local/hello-universe -> local/hello-universe:2019-10-01_12-29-07.3_IST
Tags generated in 215.418µs
Checking cache...
DEBU[0000] Found dependencies for dockerfile: [{requirements.txt /service true} {app.py /service true}]
 - local/hello-universe: Found. Tagging
Cache check complete in 12.389635ms
Tags used in deployment:
 - Since images are not pushed, they can't be referenced by digest
   They are tagged and referenced by a unique ID instead
 - local/hello-universe -> local/hello-universe:19544728a865a1bbb09f0e3e42d075d76d6fb52ee571e9d446c0841beb21abaf
Starting deploy...```

**Comment by dgageot** [comment]
@none-da This is totally expected when running with a local cluster. Skaffold has to make sure k8s is using the exact same image that you just built. The way to do that is to use an immutable reference. Tags are not immutable. So, in general, it uses the image's digest in addition to the tag. But this can't be done with a local cluster. With local clusters, images are not pushed, so we don't have their digest. Skaffold instead tags the image with a unique tag (in addition to the tag produced by the datatime policy) and uses that tag to references the image.

To make sure people understand that, it shows a warnings (that you can se in your logs):
```
Tags used in deployment:
 - Since images are not pushed, they can't be referenced by digest
   They are tagged and referenced by a unique ID instead
 - local/hello-universe -> local/hello-universe:19544728a865a1bbb09f0e3e42d075d76d6fb52ee571e9d446c0841beb21abaf
```

I'm going to close the issue since this is the expected behavior
